diff --git a/src/fw/shadow.c b/src/fw/shadow.c
index 4c627a8..504312d 100644
--- a/src/fw/shadow.c
+++ b/src/fw/shadow.c
@@ -46,8 +46,8 @@ __make_bios_writable_intel(u16 bdf, u32 pam0)
     pam[0] = 0x30;

     // Write PAM settings back to pci config space
-    pci_config_writel(bdf, ALIGN_DOWN(pam0, 4), pamdata.data32[0]);
     pci_config_writel(bdf, ALIGN_DOWN(pam0, 4) + 4, pamdata.data32[1]);
+    pci_config_writel(bdf, ALIGN_DOWN(pam0, 4), pamdata.data32[0]);

     if (!ram_present)
         // Copy bios.
diff --git a/src/hw/dma.c b/src/hw/dma.c
index 20c9fbb..f048560 100644
--- a/src/hw/dma.c
+++ b/src/hw/dma.c
@@ -57,6 +57,7 @@ dma_floppy(u32 addr, int count, int isWrite)
 void
 dma_setup(void)
 {
+  return;
     // first reset the DMA controllers
     outb(0, PORT_DMA1_MASTER_CLEAR);
     outb(0, PORT_DMA2_MASTER_CLEAR);
diff --git a/src/hw/ps2port.c b/src/hw/ps2port.c
index 9b099e8..117bea8 100644
--- a/src/hw/ps2port.c
+++ b/src/hw/ps2port.c
@@ -59,7 +59,7 @@ i8042_flush(void)
 {
     dprintf(7, "i8042_flush\n");
     int i;
-    for (i=0; i<I8042_BUFFER_SIZE; i++) {
+    for (i=0; i<0; i++) {
         u8 status = inb(PORT_PS2_STATUS);
         if (! (status & I8042_STR_OBF))
             return 0;
diff --git a/src/optionroms.c b/src/optionroms.c
index e906ab9..4427bc4 100644
--- a/src/optionroms.c
+++ b/src/optionroms.c
@@ -457,6 +457,7 @@ vgarom_setup(void)

     dprintf(1, "Scan for VGA option rom\n");

+    return;
     // Load some config settings that impact VGA.
     EnforceChecksum = romfile_loadint("etc/optionroms-checksum", 1);
     S3ResumeVga = romfile_loadint("etc/s3-resume-vga-init", CONFIG_QEMU);
diff --git a/src/serial.c b/src/serial.c
index 88349c8..3ee6ddf 100644
--- a/src/serial.c
+++ b/src/serial.c
@@ -207,6 +207,7 @@ lpt_setup(void)
     if (! CONFIG_LPT)
         return;
     dprintf(3, "init lpt\n");
+    return;

     u16 count = 0;
     count += detect_parport(PORT_LPT1, 0x14, count);
