diff --git a/src/hw/dma.c b/src/hw/dma.c
index 20c9fbb..f048560 100644
--- a/src/hw/dma.c
+++ b/src/hw/dma.c
@@ -57,6 +57,7 @@ dma_floppy(u32 addr, int count, int isWrite)
 void
 dma_setup(void)
 {
+  return;
     // first reset the DMA controllers
     outb(0, PORT_DMA1_MASTER_CLEAR);
     outb(0, PORT_DMA2_MASTER_CLEAR);
diff --git a/src/hw/ps2port.c b/src/hw/ps2port.c
index 9b099e8..117bea8 100644
--- a/src/hw/ps2port.c
+++ b/src/hw/ps2port.c
@@ -59,7 +59,7 @@ i8042_flush(void)
 {
     dprintf(7, "i8042_flush\n");
     int i;
-    for (i=0; i<I8042_BUFFER_SIZE; i++) {
+    for (i=0; i<0; i++) {
         u8 status = inb(PORT_PS2_STATUS);
         if (! (status & I8042_STR_OBF))
             return 0;
diff --git a/src/serial.c b/src/serial.c
index 88349c8..3ee6ddf 100644
--- a/src/serial.c
+++ b/src/serial.c
@@ -207,6 +207,7 @@ lpt_setup(void)
     if (! CONFIG_LPT)
         return;
     dprintf(3, "init lpt\n");
+    return;

     u16 count = 0;
     count += detect_parport(PORT_LPT1, 0x14, count);
